sql/plan/load_data.go

```diff
@@ -0,0 +1,379 @@
+		if !keepGoing {
+			return nil, io.EOF
+		}
+
+		line = l.scanner.Text()
+		exprs, err = l.parseFields(line)
+
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Match input columns with the amount of columns provided in the text.
+	// Append nils to the parsed fields if they are less than the input columns.
+	// TODO: Match schema with column order
+	colDiff := len(l.destination.Schema()) - len(exprs)
+
+	// append NULLS for the rest of the fields
+	exprs = addNullsToValues(exprs, colDiff)
+
+	// create the values that are returned as a row iter.
+	var values [][]sql.Expression
+	values = append(values, exprs)
+	newValue := NewValues(values)
+
+	ri, err := newValue.RowIter(l.ctx, returnRow)
+	if err != nil {
+		return nil, err
+	}
+
+	return ri.Next()
+}
+
+func (l loadDataIter) Close(ctx *sql.Context) error {
+	if !l.scanner.Scan() {
+		if err := l.scanner.Err(); err != nil {
+			return err
+		}
+
+		if l.local {
+			err := os.Remove(l.file.Name())
+			if err != nil {
+				return err
+			}
+		} else {
+			err := l.file.Close()
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// parseLinePrefix searches for the delim defined by linesStartingByDelim.
+func (l loadDataIter) parseLinePrefix(line string) string {
+	if l.linesStartingByDelim == "" {
+		return line
+	}
+
+	prefixIndex := strings.Index(line, l.linesStartingByDelim)
+
+	// The prefix wasn't found so we need to skip this line.
+	if prefixIndex < 0 {
+		return ""
+	} else {
+		return line[prefixIndex+len(l.linesStartingByDelim):]
+	}
+}
+
+func (l loadDataIter) parseFields(line string) ([]sql.Expression, error) {
+	// Step 1. Start by Searching for prefix if there is one
+	line = l.parseLinePrefix(line)
+	if line == "" {
+		return nil, nil
+	}
+
+	// Step 2: Split the lines into fields given the delim
+	fields := strings.Split(line, l.fieldsTerminatedByDelim)
+
+	// Step 3: Go through each field and see if it was enclosed by something
+	// TODO: Support the OPTIONALLY parameter.
+	if l.fieldsEnclosedByDelim != "" {
+		for i, field := range fields {
+			if string(field[0]) == l.fieldsEnclosedByDelim && string(field[len(field)-1]) == l.fieldsEnclosedByDelim {
+				fields[i] = field[1 : len(field)-1]
+			} else {
+				return nil, fmt.Errorf("error: field not properly enclosed")
+			}
+		}
+	}
+
+	// TODO: Step 4: Check for the ESCAPED BY parameter.
+	exprs := make([]sql.Expression, len(fields))
+
+	for i, field := range fields {
+		dSchema := l.destination.Schema()[i]
+		// Replace the empty string with defaults
+		if field == "" {
+			_, ok := dSchema.Type.(sql.StringType)
+			if !ok {
+				exprs[i] = expression.NewLiteral(dSchema.Default, dSchema.Type)
+				continue
+			}
+		}
+
+		exprs[i] = expression.NewLiteral(field, sql.LongText)
+	}
+
+	return exprs, nil
+}
+
+func addNullsToValues(exprs []sql.Expression, diff int) []sql.Expression {
+	for i := diff; i > 0; i-- {
+		exprs = append(exprs, expression.NewLiteral(nil, sql.Null))
```

Comment:
Instead of this, just make the expr array the same size as the dest schema and the remaining entries will already be `nil`